var/spack/repos/builtin/packages/py-jupyterlab-widgets/package.py

```diff
@@ -26,6 +31,7 @@ class PyJupyterlabWidgets(PythonPackage):
     )
 
     depends_on("python@3.6:", type=("build", "run"))
+    depends_on("python@3.7:", when="@3.0.3:", type=("build", "run"))
     depends_on("py-setuptools@40.8.0:", type="build")
```

Comment:
If we're installing from a wheel, I think you can remove all of these build-only dependencies